Add etcc

wannaphong opened the pull request:
Add Enhanced Thai Character Cluster (ETCC)

Notebook : https://colab.research.google.com/drive/1UTQgxxMRxOr9Jp1B1jcq1frBNvorhtBQ

wannaphong committed:
Add etcc

pep8speaks commented:
Hello @wannaphong! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:









There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2020-02-29 05:11:44 UTC

wannaphong committed:
Update etcc.py

bact reviewed (commented):
etcc.py:
- sort import
- use 4 spaces for indent
- remove trailing spaces at the end of the line
- "and" inside "if (re.search...) should be indented in

setup.py:
- sort filenames in package_data

wannaphong committed:
Update etcc.py

wannaphong commented:
Done

p16i commented:
could you please provide some unittests as well?

wannaphong committed:
Add etcc test

wannaphong commented:
> 
> 
> could you please provide some unittests as well?

Ok. I add some unittests.

bact committed:
Refactor etcc.py

- Directly pass the iterable (list of words) to the Tokenizer
- Pre-compile pattern "[ะาๆฯๅำ]"

bact committed:
segment(): Return empty list for non-text input

wannaphong committed:
Update Subword tokenizer Docs

bact committed:
Update etcc test against empty string and None

bact committed:
Update test_tokenize.py

bact committed:
Update etcc.py

bact commented:
Since it is a regular text file, does it possible to rename `etcc.dict` to `etcc.txt` ?

bact commented:
อันนี้คือเลิกใช้กฎไปเลย แต่ใช้พจนานุกรมของ etcc ที่เป็นไปได้ทั้งหมดมาใส่แทนใช่ไหมครับ

wannaphong commented:
> 
> 
> อันนี้คือเลิกใช้กฎไปเลย แต่ใช้พจนานุกรมของ etcc ที่เป็นไปได้ทั้งหมดมาใส่แทนใช่ไหมครับ

ใช่ครับ

wannaphong committed:
rename etcc.dict to etcc.txt

bact commented:
Looks nice :)

wannaphong commented:
> 
> 
> Looks nice :)

Thank you.
